### PR TITLE
Add prettier code formatting

### PR DESCRIPTION
**Pull request by jason-fox**
Adds [prettier](https://prettier.io) and  [husky](https://github.com/typicode/husky). All JavaScript files are autoformated on commit using the Git Hooks hook. 

running `npm run prettier` will format all files with standardized whitespace.

### CODE REVIEW

**Comment by fgalan** [comment]
I didn't realize looking to https://github.com/telefonicaid/iotagent-node-lib/pull/753 but it is more evident in this one.

Current code style length limit is 120, which is pretty standard nowdays, e.g default in Pycharm IDE, used by github in its file views (actually github uses 130 as far as I have checkd). Default 80 is too limitating, specially when each new intentation block "eats" 4 whispaces.

Thus, I'd suggest to add `--print-width 120` to the CLI.

Let's also include @cblanco in this conversation to know his opinion (not specifically on the line length topic but in general about this PR).

**Comment by fgalan** [comment]
@jason-fox we have just merged two big PRs (#305 and #316). So, could you update this PR with master so we can merge it?

There are other pending PR in this repo but I think they don't involve big issues:

* PR https://github.com/telefonicaid/perseo-fe/pull/309, authored by @Secmotic. I think is pretty small
* PR https://github.com/telefonicaid/perseo-fe/pull/304, not related with code

Thanks!

File: package.json [hunk 1]
```diff
@@ -23,17 +24,21 @@
     "test": "mocha --recursive 'test/**/*.js' --reporter spec --timeout 3000 --ui bdd --exit",
     "test:watch": "npm run test -- -w ./lib",
     "lint": "jshint lib/ --config .jshintrc && jshint test/ --config test/.jshintrc",
+    "prettier": "prettier --tab-width 4 --print-width 120 --single-quote --trailing-comma es5 --write ./**/**/*.js **/*.js *.js",
```
Comment (fgalan) [inline package.json]: Taking into account  .prettierrc.json I understand that `--tab-width 4 --print-width 120 --single-quote --trailing-comma es5` can be removed (as it has been done in lint-staged below).

Comment (jason-fox) [reply package.json]: Fixed a573d49

File: test/.jshintrc [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "maxparams": 6,
     "maxdepth": 4,
     "camelcase": true,
-    "maxlen": 120,
+    "maxlen": 180,
```
Comment (fgalan) [inline test/.jshintrc]: Why this change is needed?

I mean, if prettier is ensuring `"printWidth": 120`, why the linter needs to extend +60 chars the limit?

Just curiosity...

**Comment by cblanco** [comment]
LGTM. Good Work!

**Comment by fgalan** [comment]
> PR #309, authored by @Secmotic. I think is pretty small

Fortunatelly, no conflict seems to happen on #309 after merging this :)

File: test/.jshintrc (hunk 2 above)
Comment (jason-fox) [reply test/.jshintrc]: @fgalan, prettier is *attempting* to fit within 120 characters, but according to its own opinionated
rules it won't always succeed particularly if you have a line with a long string. Prettier will **always** remove unnecessary concatenations in strings, and always indent properly (with a tab 4) so if you are left with a line over 120 characters long jslint will complain.

We're fortunate that we never need very long strings in the code base then. However each of the tests **does** contain a  single lengthy  string in it - the description of the test itself - also this
ends up being prettily indented by about 4 tabs. 

Rather than "fixing" this by truncating the description text, I chose to relax the jslint rule for the **tests only**.

Of course if you leave the default tab at 2 spaces you're less likely to reach the limit.

Comment (fgalan) [reply test/.jshintrc]: I think is a good solution. No problem at all. Thanks!
